scale/recipe/models.py

```diff
@@ -157,11 +157,18 @@ def _create_recipe_jobs(self, recipe, event, when, delta, superseded_jobs):
             recipe_job.recipe = recipe
             recipe_jobs_to_create.append(recipe_job)
 
-        # Supersede any jobs that were changed or deleted in new recipe
         if delta:
-            for superseded_job_name in delta.get_deleted_nodes():
-                jobs_to_supersede.append(superseded_jobs[superseded_job_name])
+            # Go through deleted jobs, unpublish their products, and get ready to supersede them
+            from product.models import ProductFile
```

Comment:
This block should be in a try/except if you want to maintain the package hierarchy where ProductFile is optional here.